File: Phase01/test/FileReaderTest.java
package Phase01.test;

import Phase01.src.FileReader;

import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.HashMap;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;

public class FileReaderTest {
    private FileReader fileReader;
    final String PATH = "Phase01/data/Test-Data";

    @BeforeEach
    public void createObject() throws Exception {
        fileReader = new FileReader(PATH);
    }

    @Test
    public void readContentTest() {
        var contents = new HashMap<String, String>();
        contents.put("testData1.txt", "hello world..");
        contents.put("testData2.txt", "foo? bar!!!");
        assertTrue(fileReader.readContent().equals(contents));
    }

}

File: Phase05/app/library.Test/FileReaderTest.cs
using System.Collections.Generic;
using Xunit;
using System.Linq;

namespace library.Test
{
    public class FileReaderTest
    {
        private FileReader fileReader;
        const string PATH = "..\\..\\..\\..\\Test-Data";
        public FileReaderTest()
        {
            fileReader = new FileReader(PATH);
        }

        [Fact]
        public void ReadContentTest()
        {
            var contents = new Dictionary<string, string>();
            contents.Add("testData1.txt", "hello world..");
            contents.Add("testData2.txt", "foo? bar!!!");
            contents.Add("testData3.txt", "Freedom is only a hallucination.");
            Assert.True(contents.OrderBy(r => r.Key).SequenceEqual(this.fileReader.ReadContent().OrderBy(r => r.Key)));
        }
    }
}